### DISCUSSION TITLE
Set method in tests do not go through lifecycle hooks

### DISCUSSION THREAD

**Discussion by VBPdevelopment** [livewire/livewire#4541] [7. Bugs (archived)] [answered]
I have a component similar to the following:
```php
class Edit extends Component
{
    public $product;
    public $price;
    
    public function mount(Product $product): void
    {
        $this->product = $product;
        $this->price = $product->price;
    }

    public function render()
    {
        return view('livewire.product.edit');
    }

    public function save(): void
    {
        $this->validate();

        $this->product->fill([
            'price' => floatval($this->price),
        ]);
        $this->product->save();
    }

    public function updating($name, &$value): void
    {
        if (in_array('numeric', $this->getRules()[$name], true) && Str::contains($value, ',')) {
            $value = str_replace(',', '.', $value);
        }
    }

    public function rules(): array
    {
        return [
            'price' => [
                'required',
                'numeric',
                'min:0.01',
            ],
        ];
    }
}
```

I'm working on a test for this component but it's not possible to test the updating method (not directly and not using the set method in tests).

**Comment by joshhanley** [comment]
@VBPdevelopment hmm `->set()` does actually go through the lifecycle hooks, so not sure what you mean exactly.

If you can share your test, so we can have a look at what you are trying to do, and explain what you want to test, maybe we can help get it running 🙂

**Comment by VBPdevelopment** [comment]
Yes ofcourse :)

The test was very simple:

```
public function testKommaGetsReplaced(): void
{
    Livewire::test(Edit::class, [
        'product' => $product,
    ])->set('price', '13,50')->assertSet('price', '13.50');
}
```

This throws a test error.
```
Failed asserting that two strings are equal.
Expected :'13.50'
Actual   :'13,50'
```

**Comment by joshhanley** [comment]
@VBPdevelopment that test all looks good to me, so it should work as you are expecting. Maybe the test is doing its job? I'd do some dumps and double check your logic in your `updating` method is working properly 🙂

**Comment by VBPdevelopment** [comment, marked as answer]
Think i found it. Initially I did not had the value parameter by reference. This needs to be by reference to work.
This is something I did not find in the docs, might be worth adding it.

**Reply by joshhanley** [reply]
@VBPdevelopment ah interesting, I didn't comment on it because in your example code above, you were already passing by reference.

PR's are welcome to update the docs https://github.com/livewire/docs
